### PR TITLE
Added missing types for Server Options

### PR DESCRIPTION
**Pull request by nm-infy**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

Fixes: #1921

### CODE REVIEW

**Comment by nm-infy** [comment]
Formatting changes appeared when I ran `npm run lint:typescript -- --fix`.
May be we could run `npm run lint:typescript -- --fix` once to fix formatting, in a separate PR.

**Comment by nm-infy** [comment]
> There seems to be quite a few formatting changes, would you mind to roll those back?

Done. Could you please review now?

File: test/types/index.ts [hunk 1]
```diff
@@ -70,6 +70,14 @@ const cors = require('cors')
...
       }
       return Math.random()
+    },
+    requestIdHeader: 'request-id',
+    requestIdLogLabel: 'reqId',
+    serverFactory: (handler, options) => {
+      const server = http.createServer((req, res) => {
+        handler(req, res)
+      })
+      return server
```
Comment (mcollina) [inline test/types/index.ts]: Can you add a test also with http2?

**Comment by RafaelGSS** [comment]
Well, not success in node 6

**Comment by mcollina** [comment]
It’s just a flake.

**Pull request merged**